projects/rocprofiler-sdk/source/bin/rocprofv3.py

```diff
@@ -1748,36 +1748,67 @@ def main(argv=None):
         args = get_args(cmd_args, inp_args[0])
 
         if args.pid:
+            # For reattachment support, args must be the same as previous rocprofv3 sessions
+            # Store args in a temporary file for future sessions. If the temporary file already exists,
+            # compare those args and error out if the tracing options are not the same.
             import pickle
 
             if args.collection_period:
                 fatal_error("--collection-period is not compatible with attach mode")
 
             fname = f"/tmp/rocprofv3_attach_{args.pid}.pkl"
-            if os.path.exists(fname):
+            if not os.path.exists(fname):
+                # if this is the first attachment, write the temp configuration file for future attachments
+                with open(fname, "wb") as ofs:
+                    if args.log_level in ("config", "info", "trace"):
+                        print(f"Saving attach configuration to {fname}...")
+                    pickle.dump(args, ofs)
+            else:
+                # if this is not the first attachment
                 # load the configuration from the previous attachment
                 with open(fname, "rb") as ifs:
                     if args.log_level in ("config", "info", "trace"):
                         print(f"Loading attach configuration from {fname}...")
-                    prev_args = pickle.load(ifs)
-
-                args = get_args(
-                    args,
-                    dotdict(prev_args),
-                    filter=[
-                        ".*_trace",
-                        "^pc_sampling_.*$",
-                        "^att_.*$",
-                        "^(pmc|pmc_groups|output_config|extra_counters)$",
-                        "^kernel_(include_regex|exclude_regex|iteration_range)$",
-                    ],
-                )
-
-            # write the configuration for future attachments
-            with open(fname, "wb") as ofs:
-                if args.log_level in ("config", "info", "trace"):
-                    print(f"Saving attach configuration to {fname}...")
-                pickle.dump(args, ofs)
+                    previous_args = pickle.load(ifs)
+
+                # compare the previous arguments used to the current attachments's arguments
+                keys_to_ignore = [
+                    "input",
+                    "output_file",
+                    "output_directory",
+                    "output_format",
+                    "__dict__",
+                ]
```

Comment:
Please note, these are not the only flags that are allowed to be changed. Options involving kernel naming, grouping by queue, stats, summary, collection period, Perfetto, agent indexing, minimum output data, etc. can all be changed. Hence why the original did not try to generate a list of options which could change but rather a list of options that could not change